block-streamer/src/receive.rs

```diff
@@ -1,100 +1,39 @@
-            }
-        }
-
-        let mut receiving_cid = false;
-
-        loop {
-            if let Ok(len) = socket.try_recv(&mut buf) {
-                if len > 0 {
-                    let mut databuf = &buf[..len];
-                    // TODO: Some of this reassembly logic needs to be extracted so that it can be tested directly
-                    // TODO: This also should be reworked to handle a stream of chunks/blocks sent totally out of order
-                    if let Ok(message) = TransmissionMessage::decode(&mut databuf) {
-                        info!("Received Message {} bytes", len);
-                        match message {
-                            TransmissionMessage::Cid(cid) => {
-                                info!("Received CID: {:?}", cid);
-                                current_cid = Some(cid);
-                                current_cid_chunks.clear();
-                            }
-                            TransmissionMessage::Chunk(chunk) => {
-                                info!("Received Chunk for CID: {:?}", chunk.cid_marker);
-                                if let Some(current_cid) = &current_cid {
-                                    current_cid_chunks.push(chunk);
-
-                                    if let Ok(wrapper) =
-                                        BlockWrapper::from_chunks(current_cid, &current_cid_chunks)
-                                    {
-                                        let block = wrapper.to_block()?;
-                                        if !block.links().is_empty() {
-                                            info!("Found root CID {}", &block.cid());
-                                            root = Some(block);
-                                        } else {
-                                            info!(
-                                                "Found child CID {} with {} bytes",
-                                                &block.cid(),
-                                                len
-                                            );
-                                            blocks.insert(*block.cid(), block.clone());
-                                        }
-                                        receiving_cid = true;
-                                    }
-                                }
-                            }
-                        }
+        {
+            loop {
+                if let Ok(len) = socket.try_recv(&mut buf) {
+                    if len > 0 {
+                        real_len = len;
+                        break;
                     }
                 }
-            } else if receiving_cid {
-                break;
+                sleep(Duration::from_millis(10));
             }
-            sleep(Duration::from_millis(10));
+        }
+
+        let mut databuf = &buf[..real_len];
+        if let Ok(Message::DataProtocol(msg)) = Message::decode(&mut databuf) {
```

Comment:
Do we care if the decode fails?  That is, should we at least log out any decoding errors?

Reply:
Good catch, will add a log here.